app/services/pagelets/manager.rb

```diff
@@ -0,0 +1,47 @@
+module Pagelets
+  class Manager
+
+    def initialize(page_name)
+      @page_name = page_name
+    end
+
+    def add_pagelet(mountpoint, opts)
+      self.class.add_pagelet(@page_name, mountpoint, opts)
+    end
+
+    class << self
+
+      def add_pagelet(page_name, mountpoint, opts)
+        handle_empty_keys_for page_name, mountpoint
+        raise ::Foreman::Exception.new(N_("Cannot add pagelet to page %s without partial", page_name)) unless opts[:partial]
+        raise ::Foreman::Exception.new(N_("Cannot add pagelet to page %s without mountpoint", page_name)) if mountpoint.nil?
+        priority = opts[:priority] || default_pagelet_priority(page_name, mountpoint)
+        @pagelets[page_name][mountpoint] << Pagelets::Pagelet.new(opts.delete(:name), opts.delete(:partial), priority, opts)
+      end
+
+      def default_pagelet_priority(page_name, mountpoint)
+        # We need a default priority value for the first pagelet if it is not specified
+        priority = @pagelets[page_name][mountpoint].map(&:priority).max
+        priority.nil? ? 0 : priority + 100
```

Comment:
This could be more ellegant in a form of:

    @pagelets[page_name][mountpoint].map(&:priority).push(0).max